server/api/routers/projects/projects.py

```diff
@@ -293,7 +293,7 @@ async def chat(
             if record is None:
                 # Create new agent and enable persistence
                 agent = ProjectChatOrchestratorAgentFactory.create_agent(
-                    project_config, project_dir=project_dir
+                    project_config, project_dir=project_dir, model_name=request.model
```

Comment:
This is outdated with main

Reply:
This is not working as you would expect. Since, these agents are cached into the `agent_sessions` "cache", you will not create a new agent with the desired model unless ther eis a "cache miss". That will not happen on-demand by making a request with a new model. Therefore, it is likely that an existing agent will be found in the cache for this session that was created with the original or default model and this code will never run.